src/plugins/discover/public/application/angular/discover.js

```diff
@@ -444,13 +443,13 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
           allowEmbed: false,
           allowShortUrl: uiCapabilities.discover.createShortUrl,
           shareableUrl: unhashUrl(window.location.href),
-          objectId: savedSearch.id,
-          objectType: 'search',
           sharingData: {
-            ...sharingData,
-            title: savedSearch.title,
+            taskType: 'csv',
+            taskConfig: {
+              ...sharingData,
```

Comment:
`sharingData` is the return value of `searchSource.getSerializedFields`. 

Would it be possible for CSV export to use just the ID of the saved search object as the params? If so, could the searchSource object come from that ID?

If someone has automated CSV export and makes changes to the saved search after scheduling the automation, their changes should be reflected in subsequent runs of the export.

Reply:
i think we should try to get away from dependency between csv report and existing saved search. This way we can create csv report on any search made with searchsource without requirement for backing saved search object (for example export csv from visulalization, or on discover without persisting the current view)

Reply:
It would be really interesting to see how that plays out, and I think we should carry on in this direction!

There is a related issue that could be solved with this approach: https://github.com/elastic/kibana/issues/18439 (see option 3)